csharp/ql/test/library-tests/csharp9/NativeInt.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+
+public class Class1
+{
+    public void M1(int j, uint k)
+    {
+        nint x = j;
+        nint x0 = (nint)j;
+        IntPtr x1 = (IntPtr)j;
+        nuint y = k;
+
+        const nint i = (nint)42;
+    }
+
+    public void M2()
+    {
+        nint x = 3;
+        int y = 3;
+        long v = 10;
+
+        var test3 = typeof(nint);      // System.IntPtr
+        var test4 = typeof(nuint);     // System.UIntPtr
+        var test5 = (x + 1).GetType(); // System.IntPtr
+        var test6 = (x + y).GetType(); // System.IntPtr
+        var test7 = (x + v).GetType(); // System.Int64
+    }
+
+    // enum E : nint
+    // {
+    // }
```

Comment:
Remove?

Reply:
The [feature proposal](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/native-integers#miscellaneous) says that this should also work, but maybe it was not implemented or needs .net 5.

In any case, I'll remove the commented sample. It doesn't add any value.